### PR TITLE
fix: semi empirical version of experiment and add docker support for both variants

### PR DESCRIPTION
**Pull request by VassilisVassiliadis**
## Change list

- [ ] Updates description 
- [x] Updates definition of PM6 variant to use the SetSemiEmpiricalBasis component
- [x] Adds docker support for both variants

## Checklist

Ensure your PR meets the following requirements:

- [ ] This PR is associated to one (or more) issues that are open
- [x] I have signed off my commits (using `git commit -s`)
- [x] I agree with the ST4SD Code of Conduct

### CODE REVIEW

File: semi-empirical/excitation-energy-se-gamess-us.yaml [hunk 1]
```diff
@@ -141,17 +142,23 @@ components:
...
-    arguments: --input input_smiles.csv --row GetMoleculeIndex:output
+    arguments: --input input_smiles.csv 
+      --template SetSemiEmpiricalBasis/input_molecule.txt:ref
+      --row GetMoleculeIndex:output
     environment: python
     executable: bin/rdkit_smiles2coordinates.py
   references:
   - input/input_smiles.csv:copy
   - GetMoleculeIndex:output
+  - SetSemiEmpiricalBasis/input_molecule.txt:ref
```
Comment (VassilisVassiliadis) [inline semi-empirical/excitation-energy-se-gamess-us.yaml]: This uses instructs SMILESToXYZ to take into account the output of SetSemiEmpiricalBasis

**Comment by VassilisVassiliadis** [comment]
To test out this PR locally using a docker-like container runtime (docker, podman, etc) :

1. create a venv and install `pip install "st4sd-runtime-core[deploy]"`
2. activate the venv
3. clone the repository and switch to the branch `vv_fix_semi_empirical`
4. Run the virtual experiment using a docker-like container runtime

For example:

```bash
#!/usr/bin/env bash
git clone --branch vv_fix_semi_empirical https://github.com/st4sd/excitation-energy-gamess.git
cd excitation-energy-gamess

python3 -m venv --copies venv
. venv/bin/activate
pip install "st4sd-runtime-core[deploy]"

# Add 1 smiles per line
cat <<EOF >input_smiles.csv
label,smiles
0,C
EOF

lines=$(cat input_smiles.csv | wc -l)
number_molecules=$((lines-1))
cat <<EOF >variables.yaml
global:
  numberMolecules: ${number_molecules}
  docker-platform: linux/amd64
EOF

# To use podman instead of docker, set --dockerExecutableOverride podman
elaunch.py -a variables.yaml  -i input_smiles.csv  -p docker -l15 \
  --manifest semi-empirical/manifest.yaml semi-empirical/excitation-energy-se-gamess-us.yaml

# The name of the experiment is <YAML file containing virtual experiment definition>-<time stamp>.instance
most_recent=$(ls -td excitation-energy-se-gamess-us*.instance | head -1)

echo "----Experiment is complete----"
echo "Find the files that your experiment generated under ${most_recent}"
echo "For example, these are the properties that your virtual experiment computed"
cat "${most_recent}/output/properties.csv"
```

I ran the above on a Mac device with M1 and got the following properties:

```
input-id;label;excitation-energy;osc-str-max;band-gap;homo;lumo;electric-moments;total-energy
C;0;8.362;0.4906;12.302;-10.093;2.21;0.115192;-5492.829272320928
```

**Comment by michael-johnston** [comment]
@VassilisVassiliadis Did you test it locally/k8s?

**Comment by VassilisVassiliadis** [comment]
@michael-johnston I tested it locally (with `docker`) for  the input SMILES `C` and got this output:

```
input-id;label;excitation-energy;osc-str-max;band-gap;homo;lumo;electric-moments;total-energy
C;0;8.362;0.4906;12.302;-10.093;2.21;0.115192;-5492.829272320928
```

**Pull request merged**